### ISSUE TITLE
c-api

### ISSUE THREAD

**Issue by ktye** [StephenCrawley/krua#1]
do you want to participate in our [c-api](https://github.com/ktye/i/blob/master/%2B/k.h)?
ngn and i designed an api, that could be implemented by any k backend.
then extension libraries can be shared unmodified across any k implementation (written in c at least).

As your k is still young, you could implement the api directly and use the syntax internally (our K type being a struct is no problem).
Otherwise you could write glue code (as ngn and i have to do).

**Comment by StephenCrawley** [comment]
definitely. i have seen your conversation with ngn on the k tree about the api, i will take a closer look. the point of this project is for my own learning, so the fact i'm not sure what's involved for me here is a good reason to participate
